app/models/medium.rb

```diff
@@ -1263,9 +1262,31 @@ def delete_answers
       becomes(Question).answers.delete_all
     end
 
-  def answers_count
-    return -1 unless type == 'Question'
-    becomes(Question).answers.count
-  end
```

Comment:
I'm wondering why this has been a merge conflict. Have you touched this file in this PR? I haven't seen it so far.

Reply:
I was also very confused why there was a merge conflict here, as I did not touch this in the PR. How do we proceeed?

Reply:
Ok, I (probably?) figured it out: run a `git show 202635fa57` on your command line and you will see that most changes came from #495. Git could almost seamlessly merge this. However, maybe your editor indented these specific lines that are changed, so you touched them and my commit on mampf-next touched them, hence, a merge conflict arose. Maybe you could run the layout part of rubocop locally as described in #495 and see if it changes `app/models/medium.rb` again?

Reply:
Running `rubocop` autocorrection in the `app/` folder only changes one file, namely `lecture.rb`:

```
# git diff
diff --git a/app/models/lecture.rb b/app/models/lecture.rb
index a81bda89..bb7da4ee 100644
--- a/app/models/lecture.rb
+++ b/app/models/lecture.rb
@@ -803,11 +803,13 @@ class Lecture < ApplicationRecord

   def speakers
     return User.none unless seminar?
+
     User.where(id: SpeakerTalkJoin.where(talk: talks).pluck(:speaker_id))
   end

   def older_than?(timespan)
     return true unless term
+
     term.begin_date <= Term.active.begin_date - timespan
   end
```

Maybe I should revert the merge using `git revert HEAD -m 1` (?) so we can see again where the conflict lies?

Reply:
Ok, I finally pinpoined this. I just did the following locally:
```
git reset --hard origin/restrict_editor_access`
git reset --merge HEAD~1
```

so that I could replay your merge situation.

```
git checkout mampf-enxt
git pull
git checkout restrict_editor_access
git merge mampf-next
git blame -L 1279,+1 ./models/medium.rb
```

Then I saw the merge conflict in `medium.rb` and saw that you added a space in the very first commit of this PR right before the last `end` of the `medium.rb` file:
```
git blame restrict_editor_access -L 1217,+1 ./app/models/medium.rb
>> fcdcd45da (fosterfarrell9 2023-04-10 14:14:18 +0200 1217)
```

Alternatively, you directly see this in the VS Code merge editor (including `git blame` output right in the file if you have the GitLens extension enabled):
![image](https://github.com/MaMpf-HD/mampf/assets/37160523/1a84ba9f-8db5-401e-8601-f9c0948c0f55)

So that's where the merge conflict here came from. And the way you merged it is fine, so I approve this PR.


---

Bare in mind, that `git revert HEAD -m 1` might not do what is intended, see the [docs](https://git-scm.com/docs/git-revert#Documentation/git-revert.txt--mparent-number):

> Reverting a merge commit declares that you will never want the tree changes brought in by the merge. As a result, later merges will only bring in tree changes introduced by commits that are not ancestors of the previously reverted merge. This may or may not be what you want.

Also see Linus Torvald's answer [here](https://github.com/git/git/blob/master/Documentation/howto/revert-a-faulty-merge.txt), although this is not really our situation, so ``git revert HEAD -m 1` might have worked as intended.